src/backend/distributed/commands/drop_distributed_table.c

```diff
@@ -38,6 +38,8 @@ master_drop_distributed_table_metadata(PG_FUNCTION_ARGS)
 	char *schemaName = text_to_cstring(schemaNameText);
 	char *tableName = text_to_cstring(tableNameText);
 
+	EnsureSchemaNode();
```

Comment:
Minor, what about moving this to `create_reference_table ()` function given that we do it `create_distributed_table ` there?